src/shop/api.py

```diff
@@ -239,3 +245,51 @@ def assign_free_stocktake_chunk(user_id, stocktake_id):
     chunk_obj.owner_id = user_id
     chunk_obj.save()
     return chunk_obj
+
+
+@transaction.atomic
+def predict_quantity(product_id, target):
+    """Predicts when a product will reach the target quantity."""
+    product_obj = models.Product.objects.get(id=product_id)
+    if product_obj.qty <= target:
+        return date.today()
+
+    # Find the last restock transaction
+    qs = product_obj.transactions.finalized()
+    breakpoint = qs.restocks().order_by('-date_created').first()
+    if breakpoint is None:
+        return None
+
+    initial_qty = qs \
+        .filter(date_created__lte=breakpoint.date_created) \
+        .aggregate(qty=Sum('qty'))['qty'] or 0
+    trx_objs = qs \
+        .filter(date_created__gt=breakpoint.date_created) \
+        .annotate(date=TruncDay('date_created')) \
+        .values('date') \
+        .annotate(aggregated_qty=Sum('qty')) \
+        .values('date', 'aggregated_qty')
+    if not trx_objs:
+        return None
+
+    date_offset = trx_objs[0]['date'].toordinal()
+
+    x = [trx_obj['date'].toordinal() - date_offset for trx_obj in trx_objs]
+    x = [-1] + x
```

Comment:
Add a comment here, because this isn't intuitive as to why we do this.

Just to save time and headache.